icu4j/main/tests/core/src/com/ibm/icu/dev/test/util/ICUResourceBundleTest.java

```diff
@@ -1141,4 +1141,32 @@ public void TestUResourceBundleCoverage() {
         } catch (NoSuchElementException ex) {
         }
     }
+
+    @Test
+    public void TestAlgorithmicParentFallback() {
+        // Test for ICU-21125 and ICU-21126 -- cases where resource fallback isn't determined by lopping fields off
+        // the end of the locale ID (or following a %%Parent directive in a resource bundle)
+        // first column is input locale, second column is expected output locale
+        String[][] testCases = {
+            { "de_Latn_LI", "de_LI" },
+//            { "en_VA", "en_150" },  // put this back in after https://unicode-org.atlassian.net/browse/CLDR-15893 is fixed
```

Comment:
CLDR-15893 is asking to add a bunch of new English and Spanish entries to the parent-locale table.  I couldn't find an existing locale ID that would test the parent-locale-table behavior-- all of the ones that might have actual resource bundles.

Reply:
We discussed that today, and were thinking that rather than adding a bunch of empty stub locales, it would be better to add parentLocale relationships. We left this in design because we wanted to talk to you and Peter about it.

Reply:
Well, that's certainly what I was proposing-- just add entries to the `parentLocales` resource.  I know that makes that resource a lot bigger and slower to search, so maybe we'd need to change the structure in ICU to preserve performance (or size), although I don't think that'd affect CLDR.  I also know Shane has concerns about just adding a bunch of new `parentLocales` entries without proper vetting to make sure we're falling back to the right things, and I agree, provided we have a good way to get good answers to those questions.

I'm game to discuss all this whenever.